src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesCloud.java

```diff
@@ -600,7 +612,60 @@ public boolean canProvision(@NonNull Cloud.CloudState state) {
     public PodTemplate getTemplate(@CheckForNull Label label) {
         return PodTemplateUtils.getTemplateByLabel(label, getAllTemplates());
     }
+    
+    @CheckForNull
+    public PodTemplate getTemplate(@NonNull String id) {
+        return getTemplateById(id);
+    }
 
+    public PodTemplate currentTemplate;
+
+    public void doTemplates(StaplerRequest req, StaplerResponse rsp, @QueryParameter String id)
+            throws IOException, ServletException {
+        if (id != null) {
+            currentTemplate = getTemplateById(id);
+        } else {
+            currentTemplate = null;
+        }
+        req.getView(this, "templates.jelly").forward(req, rsp);
+    }
```

Comment:
Should not be necessary. `/template/{id}` should use `getTemplate(String)` method. And `templates.jelly` routing comes for free.

Reply:
It is still implemented as you describe, however, the above allows to edit the templates in the context of the cloud, which maintains the sidebar.

Reply:
the problem/challenge I have with the getTemplate(String id) As soon I want to include the sidepanel I am not able to just do <st:include page="sidepanel.jelly" it="${it}"/> without implementing a sidepanel.jelly myself, it would not highlight the templates tab in the sidepanel. If you can tell me how I could change src/main/resources/org/csanchez/jenkins/plugins/kubernetes/PodTemplate/index.jelly it so it does the same as the screenshot I am fine to remove doTemplates (edited) 
![image(2)](https://github.com/jenkinsci/kubernetes-plugin/assets/596701/04d11875-840e-4bc4-a670-398a53c1cffc)